doc/themes/scikit-learn-modern/static/js/details-permalink.js

```diff
@@ -0,0 +1,35 @@
+// Function to create permalink into <details> elements to be able to link them
+// The assumption is that such a block will be defined as follows:
+//     <details id="summary-anchor">
+//     <summary class="btn btn-light">
+//     Some title
+//     <span class="tooltiptext">Click for more details</span>
+//     <a class="headerlink" href="#summary-anchor" title="Permalink to this heading">¶</a>
+//     </summary>
+//     <div class="card">
+//     Some details
+//     </div>
+//     </details>
+// We seek to replace `#summary-anchor` with a unique identifier based on the
+// summary text.
+// This syntax is defined in `doc/conf.py` in the `rst_prolog` variable.
+function updateIdAndHrefBasedOnSummaryText() {
+    var allDetailsElements = document.querySelectorAll('details');
+
+    allDetailsElements.forEach(function (detailsElement) {
+        // Get the <summary> element within the current <details>
+        var summaryElement = detailsElement.querySelector('summary');
+
+        // The ID uses the first line, lowercased, and spaces replaced with dashes
+        var anchorID = summaryElement.textContent.trim().split("\n")[0].replace(/\s+/g, '-').toLowerCase();
+        detailsElement.setAttribute('id', anchorID);
```

Comment:
Question out of ignorance, what happens if the summary text contains special character such as `\/:*? "<>|&%. '#[]+=„` ?

Or can we rather use `.replace(/[\s\/:*?"<>|&%. '#\[\]+=„]+/g, '')`?

Reply:
I think there might be cases where the summary text is repeated across a file, such as `Examples` or similar. Is there a way to add a counter of appearance per file or allow for custom anchor in some cases?

Reply:
Regarding the special characters, I think that they can be part of the URL and it should not be a problem. I would not address this issue now. We can always change the code if in the future we have a compeling example.

For the second suggestion, this is something that I did not think about and this is a really good point. I will initiate a counter add it as a suffix.